tachyon/crypto/commitments/merkle_tree/field_merkle_tree/field_merkle_tree_mmcs.h

```diff
@@ -63,6 +70,106 @@ class FieldMerkleTreeMMCS final
     return true;
   }
 
+  const std::vector<math::RowMajorMatrix<PrimeField>>& DoGetMatrices() const {
+    return field_merkle_tree_.leaves();
+  }
+
+  [[nodiscard]] bool DoCreateOpeningProof(
+      size_t index, std::vector<std::vector<PrimeField>>* openings,
+      Proof* proof) const {
+    size_t max_rows = this->GetMaxRow();
+    size_t log_max_rows = base::bits::Log2Ceiling(max_rows);
```

Comment:
Is there an assumption where you are not using `SafeLog2Ceiling`?

Reply:
Good question! For `SafeLog2Ceiling` to be effective, we allow the rows of the matrices to be zero. I assumed that the matricies are not empty.

Reply:
Yes, exactly :)